--- a/libatomic/configure
+++ b/libatomic/configure
@@ -14859,56 +14859,6 @@ cat >>confdefs.h <<_ACEOF
 #define WORDSIZE $libat_cv_wordsize
 _ACEOF
 
-
-
-# Check to see if -pthread or -lpthread is needed.  Prefer the former.
-# In case the pthread.h system header is not found, this test will fail.
-case " $config_path " in
- *" posix "*)
-  XPCFLAGS=""
-  CFLAGS="$CFLAGS -pthread"
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-#include <pthread.h>
-      void *g(void *d) { return NULL; }
-int
-main ()
-{
-pthread_t t; pthread_create(&t,NULL,g,NULL);
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-  XPCFLAGS=" -pthread"
-else
-  CFLAGS="$save_CFLAGS $XCFLAGS" LIBS="-lpthread $LIBS"
-     cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-#include <pthread.h>
-        void *g(void *d) { return NULL; }
-int
-main ()
-{
-pthread_t t; pthread_create(&t,NULL,g,NULL);
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-
-else
-  as_fn_error $? "Pthreads are required to build libatomic" "$LINENO" 5
-fi
-rm -f core conftest.err conftest.$ac_objext \
-    conftest$ac_exeext conftest.$ac_ext
-fi
-rm -f core conftest.err conftest.$ac_objext \
-    conftest$ac_exeext conftest.$ac_ext
-  CFLAGS="$save_CFLAGS $XPCFLAGS"
-  ;;
-esac
-
 # See what sort of export controls are available.
 
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the target supports hidden visibility" >&5
diff --git a/libatomic/configure.ac b/libatomic/configure.ac
index 5563551aaae..fc60fbeb540 100644
--- a/libatomic/configure.ac
+++ b/libatomic/configure.ac
@@ -212,30 +212,6 @@ AH_BOTTOM(
 
 LIBAT_WORDSIZE
 
-# Check to see if -pthread or -lpthread is needed.  Prefer the former.
-# In case the pthread.h system header is not found, this test will fail.
-case " $config_path " in
- *" posix "*)
-  XPCFLAGS=""
-  CFLAGS="$CFLAGS -pthread"
-  AC_LINK_IFELSE(
-    [AC_LANG_PROGRAM(
-     [#include <pthread.h>
-      void *g(void *d) { return NULL; }],
-     [pthread_t t; pthread_create(&t,NULL,g,NULL);])],
-    [XPCFLAGS=" -pthread"],
-    [CFLAGS="$save_CFLAGS $XCFLAGS" LIBS="-lpthread $LIBS"
-     AC_LINK_IFELSE(
-      [AC_LANG_PROGRAM(
-       [#include <pthread.h>
-        void *g(void *d) { return NULL; }],
-       [pthread_t t; pthread_create(&t,NULL,g,NULL);])],
-      [],
-      [AC_MSG_ERROR([Pthreads are required to build libatomic])])])
-  CFLAGS="$save_CFLAGS $XPCFLAGS"
-  ;;
-esac
-
 # See what sort of export controls are available.
 LIBAT_CHECK_ATTRIBUTE_VISIBILITY
 LIBAT_CHECK_ATTRIBUTE_DLLEXPORT
diff --git a/libgomp/configure b/libgomp/configure
index 85fdb4d3f48..ee010650751 100755
--- a/libgomp/configure
+++ b/libgomp/configure
@@ -14949,48 +14949,6 @@ case "$host" in
     # NVPTX is an accelerator-only target
     libgomp_offloaded_only=yes
     ;;
-  *)
-    # Check to see if -pthread or -lpthread is needed.  Prefer the former.
-    # In case the pthread.h system header is not found, this test will fail.
-    CFLAGS="$CFLAGS -pthread"
-    cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-#include <pthread.h>
-       void *g(void *d) { return NULL; }
-int
-main ()
-{
-pthread_t t; pthread_create(&t,NULL,g,NULL);
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-  XPCFLAGS=" -Wc,-pthread"
-else
-  CFLAGS="$save_CFLAGS" LIBS="-lpthread $LIBS"
-      cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-#include <pthread.h>
-         void *g(void *d) { return NULL; }
-int
-main ()
-{
-pthread_t t; pthread_create(&t,NULL,g,NULL);
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-
-else
-  as_fn_error $? "Pthreads are required to build libgomp" "$LINENO" 5
-fi
-rm -f core conftest.err conftest.$ac_objext \
-    conftest$ac_exeext conftest.$ac_ext
-fi
-rm -f core conftest.err conftest.$ac_objext \
-    conftest$ac_exeext conftest.$ac_ext
 esac
 
 if test x$libgomp_use_pthreads != xno; then
diff --git a/libgomp/configure.ac b/libgomp/configure.ac
index a9b1f3973f7..01e6b9ae015 100644
--- a/libgomp/configure.ac
+++ b/libgomp/configure.ac
@@ -194,24 +194,6 @@ case "$host" in
     # NVPTX is an accelerator-only target
     libgomp_offloaded_only=yes
     ;;
-  *)
-    # Check to see if -pthread or -lpthread is needed.  Prefer the former.
-    # In case the pthread.h system header is not found, this test will fail.
-    CFLAGS="$CFLAGS -pthread"
-    AC_LINK_IFELSE(
-     [AC_LANG_PROGRAM(
-      [#include <pthread.h>
-       void *g(void *d) { return NULL; }],
-      [pthread_t t; pthread_create(&t,NULL,g,NULL);])],
-     [XPCFLAGS=" -Wc,-pthread"],
-     [CFLAGS="$save_CFLAGS" LIBS="-lpthread $LIBS"
-      AC_LINK_IFELSE(
-       [AC_LANG_PROGRAM(
-        [#include <pthread.h>
-         void *g(void *d) { return NULL; }],
-        [pthread_t t; pthread_create(&t,NULL,g,NULL);])],
-       [],
-       [AC_MSG_ERROR([Pthreads are required to build libgomp])])])
 esac
 
 if test x$libgomp_use_pthreads != xno; then
diff --git a/libitm/configure b/libitm/configure
index 18fc2d3a10a..13292b1ff41 100755
--- a/libitm/configure
+++ b/libitm/configure
@@ -16407,50 +16407,6 @@ $as_echo "#define AC_APPLE_UNIVERSAL_BUILD 1" >>confdefs.h
 
 # I don't like the default behaviour of WORDS_BIGENDIAN undefined for LE.
 
-
-# Check to see if -pthread or -lpthread is needed.  Prefer the former.
-# In case the pthread.h system header is not found, this test will fail.
-XPCFLAGS=""
-CFLAGS="$CFLAGS -pthread"
-cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-#include <pthread.h>
-   void *g(void *d) { return NULL; }
-int
-main ()
-{
-pthread_t t; pthread_create(&t,NULL,g,NULL);
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-  XPCFLAGS=" -Wc,-pthread"
-else
-  CFLAGS="$save_CFLAGS" LIBS="-lpthread $LIBS"
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-#include <pthread.h>
-     void *g(void *d) { return NULL; }
-int
-main ()
-{
-pthread_t t; pthread_create(&t,NULL,g,NULL);
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-
-else
-  as_fn_error $? "Pthreads are required to build libitm" "$LINENO" 5
-fi
-rm -f core conftest.err conftest.$ac_objext \
-    conftest$ac_exeext conftest.$ac_ext
-fi
-rm -f core conftest.err conftest.$ac_objext \
-    conftest$ac_exeext conftest.$ac_ext
-
 # Check for functions needed.
 for ac_func in strtoull memalign posix_memalign getauxval
 do :
